test/gpu/fuse_mlir.cpp

```diff
@@ -1069,6 +1069,136 @@ TEST_CASE(gemm_pw_softmax_gemm_pw)
     EXPECT(p1 == p2);
 }
 
+TEST_CASE(gemm_invalid_pw_softmax_gemm)
+{
+    migraphx::shape s1{migraphx::shape::half_type, {1, 12, 256, 256}};
+    auto s1_elements = s1.elements();
+
+    // Original program graph (excluding shape ops): dot -> pointwise -> softmax -> dot
+    // Here fused_reduce contains pointwise + softmax all in one module.
+    migraphx::program p1;
+    {
+        auto* mm = p1.get_main_module();
+        auto a   = mm->add_parameter("1", s1);
+        auto b   = mm->add_parameter("2", s1);
+        auto b1  = mm->add_parameter("3", s1);
+        std::vector<float> eights(s1_elements, 0.125);
+        std::vector<float> tens(s1_elements, 10);
+        auto eight = mm->add_literal(migraphx::literal{s1, eights});
+        auto ten   = mm->add_literal(migraphx::literal{s1, tens});
+        b = mm->add_instruction(migraphx::make_op("transpose", {{"permutation", {0, 1, 3, 2}}}), b);
+        b = mm->add_instruction(migraphx::make_op("contiguous"), b);
+        b1 = mm->add_instruction(migraphx::make_op("transpose", {{"permutation", {0, 1, 3, 2}}}),
+                                 b1);
+        b1 = mm->add_instruction(migraphx::make_op("contiguous"), b1);
+        auto gemm1 = mm->add_instruction(migraphx::make_op("dot"), a, b);
+
+        auto pw_reduce = add_reduce(
+            p1,
+            "main:fused_reduce0",
+            {gemm1, eight, ten},
+            {3},
+            [&](auto* rm,
+                const auto& inputs,
+                const auto& axes) -> std::vector<migraphx::instruction_ref> {
+                auto pw = add_pointwise(
+                    p1, rm, "main:pointwise0", inputs, [](auto* pm, const auto& pw_inputs) {
+                        auto mul = pm->add_instruction(
+                            migraphx::make_op("mul"), pw_inputs[0], pw_inputs[1]);
+                        return pm->add_instruction(migraphx::make_op("max"), mul, pw_inputs[2]);
```

Comment:
It would probably be good to add a pointwise op that mlir will never support(ie its only added for the tests), that way we dont need to update the test when `max` is supported.